tests/components/quick_switch_modal.test.jsx

```diff
@@ -46,65 +42,45 @@ describe('components/QuickSwitchModal', () => {
 
             wrapper.instance().handleSubmit();
             expect(baseProps.onHide).not.toBeCalled();
-            expect(props.actions.goToChannel).not.toBeCalled();
-            expect(props.actions.goToChannelById).not.toBeCalled();
-            expect(props.actions.openDirectChannelToUser).not.toBeCalled();
+            expect(props.actions.switchToChannel).not.toBeCalled();
         });
 
-        it('should open direct channel to user when selecting a dm channel', () => {
-            const props = {...baseProps};
-
+        it('should fail to switch to a channel', (done) => {
             const wrapper = shallow(
-                <QuickSwitchModal {...props}/>
+                <QuickSwitchModal {...baseProps}/>
             );
 
-            wrapper.instance().handleSubmit({channel: {id: 'channel_id', userId: 'user_id', type: Constants.DM_CHANNEL}});
-            expect(baseProps.onHide).not.toBeCalled();
-            expect(props.actions.goToChannel).not.toBeCalled();
-            expect(props.actions.goToChannelById).not.toBeCalled();
-            expect(props.actions.openDirectChannelToUser).toBeCalledWith('user_id', expect.anything(), expect.anything());
+            const channel = {id: 'channel_id', userId: 'user_id', type: Constants.DM_CHANNEL};
+            wrapper.instance().handleSubmit({channel});
+            expect(baseProps.actions.switchToChannel).toBeCalledWith(channel);
+            process.nextTick(() => {
```

Comment:
I'm not familiar with `process.nextTick`. Does it wait for all promises to resolve? It looks a bit nicer than the `Promise.resolve().then(() => {...})` that we use elsewhere